tests/core/test_open_in_process.py

```diff
@@ -106,3 +108,42 @@ async def raise_():
         with pytest.raises(InvalidDataFromChild):
             async with open_in_process(raise_) as proc:
                 await proc.wait_result_or_raise()
+
+
+@pytest.mark.trio
+async def test_timeout_waiting_for_pid(monkeypatch):
+    sleep = trio.sleep
+
+    async def wait_pid(self):
+        await sleep(constants.STARTUP_TIMEOUT_SECONDS + 0.1)
+
+    monkeypatch.setattr(Process, "wait_pid", wait_pid)
+    monkeypatch.setattr(constants, "STARTUP_TIMEOUT_SECONDS", 1)
+
+    async def do_sleep_forever():
+        while True:
+            await sleep(0.1)
```

Comment:
just in case you weren't aware, trio actually has a [`trio.sleep_forever`](https://trio.readthedocs.io/en/latest/reference-core.html#trio.sleep_forever)

Reply:
Oops, indeed. I'd used that in other tests but somehow forgot about it here